# HG changeset patch
# User Rik <rik@octave.org>
# Date 1551894681 28800
#      Wed Mar 06 09:51:21 2019 -0800
# Node ID 65c036b78040e18f8737dc17e5f0e633dc0d516c
# Parent  156bd21bf6bdd059e21c516afc0d655ab94633c8
pkg.m: Fix regression installing pkgs from URL (bug #55843).

* pkg.m: Calculate index in to files variable with strcmp.

diff --git a/scripts/pkg/pkg.m b/scripts/pkg/pkg.m
--- a/scripts/pkg/pkg.m
+++ b/scripts/pkg/pkg.m
@@ -462,17 +462,17 @@ function [local_packages, global_package
                           "This looks like an Octave Forge package name." ...
                           "  Did you mean:\n" ...
                           "       pkg install -forge %s"], ...
                          file, file);
                 else
                   error ("pkg: file not found: %s", file);
                 endif
               endif
-              files{file_idx} = local_files{end};
+              files{strcmp (files, file)} = local_files{end};
 
             endfor
           endif
         endif
         install (files, deps, prefix, archprefix, verbose, local_list,
                  global_list, global_install);
 
       unwind_protect_cleanup
